src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBulkRecordReader.java

```diff
@@ -143,19 +145,24 @@ public void close() throws IOException {
   }
 
   @VisibleForTesting
-  void setupParser(InputStream queryResponseStream) throws IOException {
+  void setupParser(String jobId, String batchId, String[] resultIds) throws IOException, AsyncApiException {
     CSVFormat csvFormat = CSVFormat.DEFAULT.
       withHeader().
       withQuoteMode(QuoteMode.ALL).
       withAllowMissingColumnNames(false);
 
-    csvParser = CSVParser.parse(queryResponseStream, StandardCharsets.UTF_8, csvFormat);
+    for (String resultId: resultIds) {
+      InputStream queryResponseStream = bulkConnection.getQueryResultStream(jobId, batchId, resultId);
+      CSVParser csvParser = CSVParser.parse(queryResponseStream, StandardCharsets.UTF_8, csvFormat);
 
-    if (csvParser.getHeaderMap().isEmpty()) {
-      throw new IllegalStateException("Empty response was received from Salesforce, but csv header was expected.");
+      if (csvParser.getHeaderMap().isEmpty()) {
+        throw new IllegalStateException("Empty response was received from Salesforce, but csv header was expected.");
+      }
+      csvParsers.add(csvParser);
     }
 
-    parserIterator = csvParser.iterator();
+    parserIterator = Iterators.concat(csvParsers.stream().map(CSVParser::iterator)
+                                        .collect(Collectors.toList()).iterator());
   }
 
   /**
```

Comment:
@return should be updated in this javadoc

Reply:
fixed